pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -39,59 +39,79 @@ func createRouteStatus(gateways []routeParentReference, obj config.Config, curre
 	}
 	// Collect all of our unique parent references. There may be multiple when we have a route without section name,
 	// but reference a parent with multiple sections.
+	// While we process these internally for-each sectionName, in the status we are just supposed to report one merged entry
 	seen := map[k8s.ParentReference]routeParentReference{}
 	failedCount := map[k8s.ParentReference]int{}
-	for _, gw := range gateways {
+	successCount := map[k8s.ParentReference]int{}
+	for _, incoming := range gateways {
 		// We will append it if it is our first occurrence, or the existing one has an error. This means
 		// if *any* section has no errors, we will declare Admitted
-		if gw.DeniedReason != nil {
-			failedCount[gw.OriginalReference]++
+		if incoming.DeniedReason != nil {
+			failedCount[incoming.OriginalReference]++
+		} else {
+			successCount[incoming.OriginalReference]++
+		}
+		exist, f := seen[incoming.OriginalReference]
+		if !f {
+			exist = incoming
+		} else {
+			if incoming.DeniedReason == nil {
+				// The incoming gateway has no error; wipe out any errors
+				// When we have multiple, this means we attached without sectionName - we only need one success to be valid.
+				exist.DeniedReason = nil
+			} else if exist.DeniedReason != nil {
```

Comment:
With `istiomultierror.New()` you could also handle reason.

Reply:
Not sure how - I think this needs to go into the status, as message string, not an issue of returning an error in code.